watertap/property_models/activated_sludge/tests/test_asm2d_reaction.py

```diff
@@ -0,0 +1,628 @@
+
+        m.fs = FlowsheetBlock(default={"dynamic": False})
+
+        m.fs.props = ASM2dParameterBlock()
+        m.fs.rxn_props = ASM2dReactionParameterBlock(
+            default={"property_package": m.fs.props}
+        )
+
+        m.fs.R1 = CSTR(
+            default={
+                "property_package": m.fs.props,
+                "reaction_package": m.fs.rxn_props,
+            }
+        )
+
+        iscale.calculate_scaling_factors(m.fs)
+
+        # NOTE: Concentrations of exactly 0 result in singularities, use EPS instead
+        EPS = 1e-8
+
+        # Feed conditions based on manual mass balance of inlet and recycle streams
+        m.fs.R1.inlet.flow_vol.fix(18446 * units.m**3 / units.day)
+        m.fs.R1.inlet.temperature.fix(298.15 * units.K)
+        m.fs.R1.inlet.pressure.fix(1 * units.atm)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_O2"].fix(EPS * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_N2"].fix(EPS * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_NH4"].fix(16 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_NO3"].fix(EPS * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_PO4"].fix(3.6 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_F"].fix(30 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_A"].fix(20 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "S_I"].fix(30 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_I"].fix(25 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_S"].fix(125 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_H"].fix(30 * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_PAO"].fix(EPS * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_PP"].fix(EPS * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_PHA"].fix(EPS * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_AUT"].fix(EPS * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_MeOH"].fix(EPS * units.mg / units.liter)
+        m.fs.R1.inlet.conc_mass_comp[0, "X_MeP"].fix(EPS * units.mg / units.liter)
+        # No data on TSS from EXPOsan at this point
+        # However, TSS is needed for this reaction
+        m.fs.R1.inlet.conc_mass_comp[0, "X_TSS"].fix(100 * units.mg / units.liter)
+
+        # Alkalinity was givien in mg/L based on C
+        m.fs.R1.inlet.alkalinity[0].fix(61 / 12 * units.mmol / units.liter)
+
+        m.fs.R1.volume.fix(1000 * units.m**3)
+
+        return m
+
+    @pytest.mark.component
+    def test_dof(self, model):
+        assert degrees_of_freedom(model) == 0
+
+    @pytest.mark.component
+    def test_unit_consistency(self, model):
+        assert_units_consistent(model) == 0
+
+    @pytest.mark.component
+    def test_solve(self, model):
+        model.fs.R1.initialize(optarg={"bound_push": 1e-8, "mu_init": 1e-8})
```

Comment:
does model have trouble initializing/solving without these solver options? I forget what the defaults are for ipopt-watertap...not sure, if they are the same as what you have here. @bknueven do you recall what the defaults are off the top of your head? I know we can find these values plugins probably.

Reply:
The default for `bound_push` is `1e-2` and for `mu_init` is `1e-1`.

I would prefer we avoid custom options in watertap, but in this particular case these make sense if you essentially expect to be at the solution already and just want to verify it.